src/effect.rs

```diff
@@ -452,28 +460,18 @@ pub fn validate_effect_control(
     #[cfg(feature = "ck3")]
     if Game::is_ck3() && (caller == "send_interface_message" || caller == "send_interface_toast") {
         vd.field_item("type", Item::Message);
-        vd.field_validated_sc("title", sc, validate_desc);
-        vd.field_validated_sc("desc", sc, validate_desc);
-        vd.field_validated_sc("tooltip", sc, validate_desc);
-        let icon_scopes =
-            Scopes::Character | Scopes::LandedTitle | Scopes::Artifact | Scopes::Faith;
-        if let Some(token) = vd.field_value("left_icon") {
-            validate_target_ok_this(token, data, sc, icon_scopes);
-        }
-        if let Some(token) = vd.field_value("right_icon") {
-            validate_target_ok_this(token, data, sc, icon_scopes);
-        }
         if let Some(token) = vd.field_value("goto") {
             let msg = "`goto` was removed from interface messages in 1.9";
             warn(ErrorKey::Removed).msg(msg).loc(token).push();
         }
-        // These seem to be scopes to set for the message loca.
-        vd.field_validated_block("localization_values", |block, data| {
-            let mut vd = Validator::new(block, data);
-            vd.unknown_value_fields(|_key, value| {
-                validate_target_ok_this(value, data, sc, Scopes::all());
-            });
-        });
+        // Mark all these as known fields. They will be validated after the main loop,
+        // because the main loop may set named scopes for them to use.
+        vd.field("title");
+        vd.field("desc");
+        vd.field("tooltip");
+        vd.field("left_icon");
+        vd.field("right_icon");
+        vd.field("localization_values");
```

Comment:
Is it necessary to be known fields since they are checked later?

Reply:
Yes because the main loop traverses unknown fields. Marking them here will except them from regular effects processing.

Reply:
I clarified the comment in the rebase